### DISCUSSION TITLE
Is there any way to generate copy button and line number when using  rehype-highlight?

### DISCUSSION THREAD

**Discussion by ideaviewes** [rehypejs/rehype#99] [Q&A] [answered]
Is there any way to generate copy button and line number when using  rehype-highlight?

**Comment by ChristianMurphy** [comment, marked as answer]
If you are also using react https://github.com/mdx-js/mdx/discussions/1948 may be of interest

**Reply by ideaviewes** [reply]
ok，thank you

**Comment by hojas** [comment]
I am using ts, how to generate a copy button with rehype-highlight?

**Reply by Murderlon** [reply]
You could take inspiration from https://github.com/s0/remark-code-extra
> For example, you can add dedicated links to code snippets that copy the code, or link to a live demo etc...
